src/askvm/resources/core/assign.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  any,
+  resource,
+  run,
+  runUntyped,
+  Options,
+  TypedValue,
+  JSONable,
+} from '../../lib';
+
+function assignValue(
+  valueObject: { [key: string]: any },
+  key: string,
+  assignedValue: any
+) {
+  if (Object.isFrozen(valueObject[key]))
+    throw new Error(`Cannot assign to a constant variable "${key}"`);
```

Comment:
we don't need this check because `assign` can only change value of a variable already created

Reply:
> we don't need this check because `assign` can only change value of a variable already created

What about the case when a const was already created and I use assign to give it new value?
From what I understand, the resource assign is used when I type equals sign '=', do I understand it correctly?

Reply:
So in a code like this, the assign resource is used 1 time or 2 times?
asl{
`const coco = 8`
 `coco = 9` -> if (Object.isFrozen(valueObject[key])) should be checked here?
}

Reply:
@mhagmajer - this is needed. The only difference between const and let is the object freeze.
@dastin-sandura - this should have a test in `assign.test.ts`.

Reply:
@mhagmajer this is the last unresolved comment from you 🙂
I have implemented and pushed the rest of changes that you have commented on so this is the last % 😄